src/plugins/intel_gpu/src/plugin/ops/range.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <intel_gpu/plugin/program.hpp>
+#include <intel_gpu/plugin/common_utils.hpp>
+
+#include <intel_gpu/primitives/range.hpp>
+#include <ngraph/op/range.hpp>
+
+namespace ov {
+namespace runtime {
+namespace intel_gpu {
+
+static void CreateRangeOp(Program& p,
+                          const std::shared_ptr<ngraph::op::v4::Range>& op) {
+  p.ValidateInputs(op, {3});
+  auto& outShape = op->get_output_partial_shape(0);
+  auto outRank = outShape.rank();
+  {
+    auto r = outRank.get_length();
+    if (r != 1)
+      throw std::runtime_error{"range v4 output rank is " + std::to_string(r)};
+  }
+  cldnn::tensor outTensor {tensor_from_dims(op->get_output_shape(0))};
```

Comment:
This will create such tensor in case of `rank== 1`: `cldnn::tensor(cldnn::batch(dims[0]), cldnn::feature(def), cldnn::spatial(def, def)) `
But your kernel expects data in other dimensions (`zyx`) - this behavior should be aligned

Reply:
i think kernel expects larger rank because there's no way to have rank 1 in cldnn. semantically range output is rank 1, i.e. all additional dimensions have length 1

Reply:
Here I'm talking mostly about configuring gws sizes as:
`dispatchData.gws[2] = params.output.Z().v * params.output.Y().v * params.output.X().v`
And using it in kernel as:
`const uint i = get_global_id(2);`

And I want to ensure that `tensor_from_dims()` will return correct tensor with single value in correct dimension. Because I have a feeling that it will create a tensor with sizes [N, 1, 1, 1] (in bfyx format) instead of expected [1, 1, 1, N] for N requested elements - which will lead to incorrect gws sizes and results. Could you please check it?

Reply:
it's not easy to check since this code isn't being run from end to end currently, but i see what you mean.
reworked